module-system/sov-modules-api/src/gas.rs

```diff
@@ -0,0 +1,160 @@
+pub mod sov_modules_api {
+    use anyhow::Result;
+
+    use core::marker::PhantomData;
+
+    // We add `remainder_gas` to `WorkingSet` as a simple `u64`; the gas will already be converted
+    // to scalar value when charged from `WorkingSet`.
+
+    pub struct WorkingSet<C> {
+        remainder_gas: u64,
+        _context: PhantomData<C>,
+    }
```

Comment:
we can add also gas_price: GasUnit and together with reminding_funds put it in GasoMeter